### PR TITLE
Add Dell E7240 Profile

### PR DESCRIPTION
**Pull request by ashgillman**
Current XPS 15 profile works well for Dell E7240 ultrabook.

### CODE REVIEW

File: dell/e7240/default.nix [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{ lib, ... }:
+
+{
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+  ];
+
+  boot.loader.systemd-boot.enable = lib.mkDefault true;
+  boot.loader.efi.canTouchEfiVariables = lib.mkDefault true;
```
[the remark this answers is not in the document]
Comment (ashgillman) [reply dell/e7240/default.nix]: Hmm, not sure. I replace everything in the generator so was useful for me. Also, this is in the XPS13 setup. So I think it's best to have it here, but if you disagree I wouldn't argue against it.

[the remark this answers is not in the document]
Comment (Mic92) [reply dell/e7240/default.nix]: The second line is something one might not want to have by default enabled: https://github.com/NixOS/nixos-hardware/pull/68#discussion_r206080663

[the remark this answers is not in the document]
Comment (ashgillman) [reply dell/e7240/default.nix]: Good call.

But still, what about the first. I assume you do actually still want this generally for this laptop?

**Comment by ashgillman** [comment]
So I added pulseaudio and I have found that to fix a bug with suspending, you need to use an older kernel version. Only, the available kernel versions change from nixpkgs to nixpkgs, so this may not be actually desirable to specify.

**Comment by ashgillman** [comment]
So I added pulseaudio and I have found that to fix a bug with suspending, you need to use an older kernel version. Only, the available kernel versions change from nixpkgs to nixpkgs, so this may not be actually desirable to specify.

**Comment by danbst** [comment]
Agree, too opinionated IMO. The problem "doesn't work on kernel XXX" should be solved in a more generic way. Maybe just write a note in README: "if you experience problems with suspend, try setting older (4.14) kernel". It may also be wrong to use old kernel next year - fix may be upstreamed already.

@ashgillman what do you think?

**Comment by ashgillman** [comment]
I think all previous comments are fair. I have updated, moving kernel choice to a comment.

File: README.org [hunk 2]
```diff
@@ -24,6 +24,7 @@ should look like:
 | Apple MacBook Air 6,X     | ~<nixos-hardware/apple/macbook-air/6>~     |
 | [[file:apple/macbook-pro/10-1][Apple MacBook Pro 10,1]]    | ~<nixos-hardware/apple/macbook-pro/10-1>~  |
 | Apple MacBook Pro 12,1    | ~<nixos-hardware/apple/macbook-pro/12-1>~  |
+| Dell E7240                | ~<nixos-hardware/dell/e7240>~              |
```
Comment (danbst) [inline README.org]:
```suggestion
| Dell E7240 [[file:dell/e7240/README.md][(notes)]]               | ~<nixos-hardware/dell/e7240>~              |
```

**Comment by Mic92** [comment]
Was merged here: https://github.com/NixOS/nixos-hardware/pull/122/files

**Pull request closed without merging**